src/main/java/com/google/api/codegen/transformer/csharp/CSharpGapicClientPackageTransformer.java

```diff
@@ -0,0 +1,140 @@
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class CSharpGapicClientPackageTransformer implements ModelToViewTransformer<ProtoApiModel> {
+  private static final String CSPROJ_TEMPLATE_FILENAME = "csharp/gapic_csproj.snip";
+
+  private static final CSharpAliasMode ALIAS_MODE = CSharpAliasMode.Global;
+
+  private final GapicCodePathMapper pathMapper;
+  private final PackageMetadataConfig packageMetadataConfig;
+  private final CSharpCommonTransformer csharpCommonTransformer = new CSharpCommonTransformer();
+  private final FileHeaderTransformer fileHeaderTransformer =
+      new FileHeaderTransformer(new StandardImportSectionTransformer());
+  private final PackageMetadataTransformer metadataTransformer = new PackageMetadataTransformer();
+
+  public CSharpGapicClientPackageTransformer(
+      GapicCodePathMapper pathMapper, PackageMetadataConfig packageMetadataConfig) {
+    this.pathMapper = pathMapper;
+    this.packageMetadataConfig = packageMetadataConfig;
+  }
+
+  @Override
+  public List<ViewModel> transform(ProtoApiModel model, GapicProductConfig productConfig) {
+    List<ViewModel> surfaceDocs = new ArrayList<>();
+    SurfaceNamer namer = new CSharpSurfaceNamer(productConfig.getPackageName(), ALIAS_MODE);
+    CSharpFeatureConfig featureConfig = new CSharpFeatureConfig();
+
+    InterfaceModel lastApiInterface = null;
+    for (InterfaceModel apiInterface : model.getInterfaces()) {
+      lastApiInterface = apiInterface;
+    }
+
+    GapicInterfaceContext context =
+        GapicInterfaceContext.create(
+            lastApiInterface,
+            productConfig,
+            csharpCommonTransformer.createTypeTable(productConfig.getPackageName(), ALIAS_MODE),
+            namer,
+            featureConfig);
+    surfaceDocs.add(generateCsProjView(context));
+
+    return surfaceDocs;
+  }
+
+  @Override
+  public List<String> getTemplateFileNames() {
+    return Arrays.asList(CSPROJ_TEMPLATE_FILENAME);
+  }
+
+  private PackageInfoView generateCsProjView(GapicInterfaceContext context) {
+    Model model = context.getModel();
+    GapicProductConfig productConfig = context.getProductConfig();
+    PackageInfoView.Builder view = PackageInfoView.newBuilder();
+    view.templateFileName(CSPROJ_TEMPLATE_FILENAME);
+    String outputPath =
+        pathMapper.getOutputPath(context.getInterface().getFullName(), productConfig);
+    view.outputPath(outputPath + File.separator + productConfig.getPackageName() + ".csproj");
+    view.fileHeader(fileHeaderTransformer.generateFileHeader(context));
+    view.serviceTitle(model.getServiceConfig().getTitle());
+    view.serviceDescription(model.getServiceConfig().getDocumentation().getSummary().trim());
+    view.domainLayerLocation(productConfig.getDomainLayerLocation());
+    view.authScopes(new ArrayList<>()); // Unused in C#
+    view.releaseLevel(productConfig.getReleaseLevel());
+    String versionSuffix;
```

Comment:
Nit: Replace mutable variable with private method.

Reply:
This method was moved from https://github.com/googleapis/gapic-generator/blob/master/src/main/java/com/google/api/codegen/transformer/csharp/CSharpGapicClientTransformer.java#L160 , and I made no modifications to it - would you be ok if I left it as-is? (applies to the two other comments you left also)

Reply:
I'm a big fan of the Boy Scout Rule and depending on the tests (and code reviewers) to catch regression. (Doc forthcoming.) YMMV though. Since it doesn't make the codebase *worse* then I guess it's fair enough.

Reply:
That approach only works if we make an explicit note (e.g. issue, doc) that we will indeed come back later and pick up all the cigarette butts.